airflow/providers/common/sql/operators/sql.py

```diff
@@ -60,6 +74,12 @@ def _get_failed_checks(checks, col=None):
     ]
 
 
+def _raise_exception(exception_string, retry_on_failure):
```

Comment:
```suggestion
def _raise_exception(exception_string: str, retry_on_failure: bool) -> NoReturn:
```

NoReturn should be imported from typing.